### DISCUSSION TITLE
save current layout

### DISCUSSION THREAD

**Discussion by kinghat** [zellij-org/zellij#2282] [Ideas]
ability to save your currently configured layout/session

**Comment by Mehdi-YC** [comment]
please read this : 
https://zellij.dev/documentation/layouts.html

**Reply by kinghat** [reply]
im aware you can manually write it out. im talking about the ability to just save the layout you have already setup on the frontend.

**Comment by Mehdi-YC** [comment]
if it is a simple layout , you can write it manually pretty easily , else il not sure if it is possible sorry.

**Comment by DumboJetEngine** [comment]
I wanted this feature too.
If it is hard to implement it all the way, saving only the tabs and panes layout, and leaving the rest for us to add (commands per pane, arguments, cwd, etc) would be fine too.
If you are to write a layout containing 7 tabs, with 3-4 panes each, and each with its own size configuration, it is not so easy...
It would nice to have something to start with.
Plus, it would help speed up the process for new users that have not read the documentation.

**Comment by Lockszmith-GH** [comment]
Look at [Session Resurrection](https://zellij.dev/documentation/session-resurrection.html?highlight=cache#session-files-in-the-cache) - the latest actual layouts are cached there in a layout file you can copy and 'abuse'.

**Reply by DumboJetEngine** [reply]
Right!
All cached sessions are here:
```
~/.cache/zellij/<version>/sesion_info/<session-name>.kdl
```
Thanks! :)

**Comment by imsnif** [comment]
You can also do this manually with `zellij action dump-layout` and get the current session's layout in STDOUT.

**Reply by DumboJetEngine** [reply]
Thanks. :)
My version (0.38.2) does not seem to support this, but I should probably upgrade.

**Reply by dievilz** [reply]
This is what I was looking for too. Someone should set this as the answer.
